UnitTests/Input/KeyTests.cs

```diff
@@ -337,4 +329,15 @@ public void TryParse_ShouldReturnTrue_WhenValidKey (string keyString, Key expect
 	[InlineData ("0x99")]
 	[InlineData ("Ctrl-Ctrl")]
 	public void TryParse_ShouldReturnFalse_On_InvalidKey (string keyString) => Assert.False (Key.TryParse (keyString, out var _));
+
+	[Theory]
+	[InlineData (KeyCode.A, KeyCode.A)]
+	[InlineData (KeyCode.F1, KeyCode.F1)]
+	public void Casting_Between_Key_And_KeyCode (KeyCode keyCode, KeyCode key)
+	{
+		Assert.Equal (keyCode, (Key)key);
+		Assert.NotEqual (keyCode, ((Key)key).WithShift);
```

Comment:
But KeyCode.A is equal to Key.A which represents the lower case without shift. So both are lower case. I think it's correct.

Reply:
Before the `Key.A` (now `KeyCode`) was considered as a capital letter `'A'` because the `Key.ShiftMask` was omitted. Now it's accomplish as `KeyCode.A | KeyCode.ShiftMask` to represent the capital `A` character. I think this change confuse you.

Reply:
Quite possibly. And it may also only matter on Windows.

Reply:
Yes, on CursesDriver and NetDriver we have to map the modifiers because they only provide character and ANSI escape, to be consistent on all drivers.

Reply:
`ConsoleKeyInfo` only handles the `shift`, `alt` and `control` because it use `ConsoleModifiers` which support only theses three modifiers. So the `capslock`  only can be handles by the `WindowsDriver` which use the `ConsoleKeyInfoEx` that store more the capslock, numlock and scrolllock. For this reason there is no sense to make unit tests that handles with this extensions modifiers because `dotnet` doesn't provide library for them, only Win32 API.